src/data/complex/module.lean

```diff
@@ -32,9 +34,48 @@ noncomputable theory
 
 namespace complex
 
-instance algebra_over_reals : algebra ℝ ℂ := (complex.of_real).to_algebra
+variables {R : Type*} {S : Type*}
 
-@[simp] lemma smul_coe {x : ℝ} {z : ℂ} : x • z = x * z := rfl
+section
+
+variables [has_scalar R ℝ]
+
+instance : has_scalar R ℂ :=
+{ smul := λ r x, ⟨r • x.re, r • x.im⟩ }
+
+lemma smul_re' (r : R) (z : ℂ) : (r • z).re = r • z.re := rfl
```

Comment:
Since this is a generalization of `smul_re` that apparently does not require any additional work to use, could you call this definition `smul_re` and the original `real_smul_re` or something like that?

Reply:
Bizarrely the original `smul_re` is not a statement about `smul` at all.

Reply:
You are right, unfortunately! Then definitely the previous should be something like `coe_smul_re`.

Reply:
Updated